juju/model.py

```diff
@@ -1469,9 +1469,21 @@ async def create_backup(self, notes=None, keep_copy=False, no_download=False):
         """
         backups_facade = client.BackupsFacade.from_connection(self.connection())
         results = await backups_facade.Create(notes=notes, keep_copy=keep_copy, no_download=no_download)
+
         if results is None:
-            raise JujuAPIError("Couldn't create a backup")
-        return results.serialize()
+            raise JujuAPIError("unable to create a backup")
+
+        backup_metadata = results.serialize()
+
+        if 'error' in backup_metadata:
+            raise JujuBackupError("unable to create a backup, got %s from Juju API" % backup_metadata)
+
+        backup_id = backup_metadata['id']
+
+        if not no_download:
+            await self.download_backup(backup_id)
+
+        return backup_metadata
```

Comment:
I'm a little confused:

- we are returning a filename (`backup_metadata['filename']`), along with a lot more information that can be used about the created backup.
- I don't know how `keep_copy` and `no_download` used to work before. Currently they seem to be doing what they intend to do, so I'm not sure why would you want to remove currently working funcionality, since we do seem have those arguments defined on Juju (unless we're dropping them soon).